### ISSUE TITLE
Add descriptions or some metadata on the Flow mode component graph

### ISSUE THREAD

**Issue by gavinwoolley** [grafana/agent#6286]
### Request

If the graph on Flow mode UI had somewhere we could display a brief description or other value, it might help describe what each component does and why

### Use case

Help myself and team understand the flow mode components, configuration and use better at a glance.

**Comment by ptodev** [comment]
Hello, thank you for raising an issue! Do you think a link to component reference docs would suffice? E.g. a link to [prometheus.scrape](https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.scrape/), from the UI?

**Comment by gavinwoolley** [comment]
Yes i think having a link to the component docs would be handy too
I was thinking each component could have an optional `metadata` or `description` argument and it could display a comment in the UI like this . 
![image](https://github.com/grafana/agent/assets/13328795/fcc81d0a-13d3-434a-9e35-22dfbe4e0d42)

**Comment by ptodev** [comment]
Thanks for clarifying! I see what you mean. Initially I thought the request was for a generic description which is the same for each component of a certain type. However, the request is actually for a description of this particular instance of the component.

Normally we'd hope that the component name is descriptive enough for this purpose. I do agree that the proposed feature could be nice. If the text is long, then the box could expand to fill it. There could also be a way to hide the descriptions in the UI so that people can see a more compact mode.

IMO the main question is how to add this description in the River config. Instead of using a component argument, it'd be cool if it actually displays the contents of a comment above the component definition. People already tend to document components like this, and they'd get it for free in this case.

**Comment by gavinwoolley** [comment]
yeh your absolutely right, my river config contains lots of `//this does this` and `this does that` comments
its them i wanted exposing 
![image](https://github.com/grafana/agent/assets/13328795/75cdd6d7-de0c-4c3d-a4b1-51c25f0a4e70)
